### PR TITLE
Jwoo/ios custom rendering part3

### PR DESCRIPTION
**Pull request by nesalang**
final addition to custom rendering
extended custom rendering to all renderer

### CODE REVIEW

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfig.h [hunk 1]
```diff
@@ -13,4 +13,12 @@
 - (instancetype)init;
 
 + (ACOHostConfigParseResult *)fromJson:(NSString *)payload;
+
 @end    
+
+
+@interface ACOTextConfig:NSObject
```
Comment (bviglietta) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfig.h]: What's this? It doesn't look like it's implemented anywhere.

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfig.mm [hunk 2]
```diff
@@ -19,15 +20,17 @@ @implementation ACOHostConfig
 
 - (instancetype)init
 {
-    self = [super init];
+    self = [self initWithConfig:nil];
```
Comment (bviglietta) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfig.mm]: >nil [](start = 32, length = 3)

nullptr

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfigPrivate.h [hunk 3]
```diff
@@ -6,13 +6,24 @@
...
+#import "TextBlock.h"
+#import <UIKit/UIKit.h>
 
 using namespace AdaptiveCards;
 
 @interface ACOHostConfig()
 
 - (instancetype)initWithConfig:(std::shared_ptr<HostConfig> const &)config;
-- (std::shared_ptr<HostConfig>)getHostConfig;
+- (std::shared_ptr<HostConfig> const &)getHostConfig;
```
Comment (bviglietta) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfigPrivate.h]: >const & [](start = 31, length = 7)

Why did you change this to return by reference? That seems unsafe.

**Comment by bviglietta** [comment]
Please don't send the same changes in two different pull requests.

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfig.h (hunk 1 above)
Comment (nesalang) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfig.h]: thank you very much

---
In reply to: [171735944](https://github.com/Microsoft/AdaptiveCards/pull/1178#discussion_r171735944) [](ancestors = 171735944)

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfigPrivate.h (hunk 3 above)
Comment (nesalang) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfigPrivate.h]: thanks

---
In reply to: [171736471](https://github.com/Microsoft/AdaptiveCards/pull/1178#discussion_r171736471) [](ancestors = 171736471)

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfig.mm (hunk 2 above)
Comment (nesalang) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOHostConfig.mm]: thanks

---
In reply to: [171736222](https://github.com/Microsoft/AdaptiveCards/pull/1178#discussion_r171736222) [](ancestors = 171736222)
